content/waku/2024-gantt.md

```diff
@@ -0,0 +1,98 @@
+# Waku Roadmap 2024 Gantt Chart
+
+Status short term work only:
+
+- reliability for 1:1 chat and communities
+- up to 100 communities
+
+Colour legend:
+
+- **Red**: engineering work to deliver the feature.
+- Other: test and telemetry work to ensure quality
+
+golang engineers: working on status-go/go-waku
+nim engineers: working on nwaku, may be a CC from research team too.
+
+```mermaid
+gantt
+    dateFormat YYYY-MM-DD
+    axisFormat %d-%b
+    
+%% Milestones overview with deliverables
+    section Store Service Upgrade
+        Store v3-beta (msg hash): crit, milestone, after storev3-br, 0
+        Store v3 (sync): crit, milestone, after storev3-r storev3-g storev3-n, 0
+        DoS protection for req-res protocols: crit, milestone, after dosreqresn dosreqresj dosreqresg, 0
+        PostgreSQL maintenance: crit, milestone, after pgsql, 0
+        (metric) Count store messages: milestone, after count-store-msg, 0
+    section Direct Message Reliability
+        (testing) direct messages: milestone, after test-direct-msg, 0
+        Review connection management: crit, milestone, after review-conn-mgmt-1 review-conn-mgmt-2, 0
+        (telemetry) Fleet logging: milestone, after telem-fleet-logging, 0
+        (telemetry) direct message reliability: milestone, after telem-d-msg-rel, 0
+        Reliability Protocol for Relay: crit, milestone, after rel-relay-g rel-relay-n, 0
+        Reliability Protocol for Resource-Restricted Clients: crit, milestone, after rel-reqres-g rel-reqres-j, 0
+        Review MVDS usage and fail path: crit, milestone, after mvds, 0
+    section Static Sharding - dedicated shards
+        (telemetry) Measure Bandwidth: milestone, after telem-bandwidth, 0
+        Sharding peer mgmt and discovery hardening: crit, milestone, after sh-peer-mgmt-j sh-peer-mgmt-n sh-peer-mgmt-g, 0
+        (testing) Custom shard impl of Communities: milestone, after test-custom-shard, 0
+    
+    section golang eng 1
+        %% Estimation TBC
+        Review connection management: crit, review-conn-mgmt-1, 2024-05-13, 8w
+        %% Estimate TBC
+        Review MVDS usage and fail path: crit, mvds, after review-conn-mgmt-1, 6w
+        %% Estimate/start TBC
+        (testing) direct messages: test-direct-msg, after mvds, 4w
```

Comment:
> I think we should provide a complete story with peer management only relates to go-waku.

sorry not sure I got that. Are you saying we should have a separate deliverable for peer management in go-waku?

I have updated `Review connection management` in roadmap doc to include status-go and go-waku.
File if it's all in go-waku and cli status-go level.

Reply:
As of now only relay peers are managed by go-waku. All other protocol specific peers are managed in status-go i.e store, peer-exchange etc. 
If we want to test these, then this will have to be done by CLI (above these components) and not just go-waku.

Reply:
> All other protocol specific peers are managed in status-go i.e store, peer-exchange etc.

From the call we just had with @arnetheduck, we should consider moving those to go-waku.
I'd say, this is part of the review connection management deliverable.